squire_lib/src/ffi_tournament.rs

```diff
@@ -53,41 +57,75 @@ impl TournamentId {
         return ptr;
     }
 
+    /// Adds a player to a tournament
+    /// On error a NULL UUID is returned
+    #[no_mangle]
+    pub unsafe extern "C" fn tid_add_player(self: Self, __name: *const c_char) -> PlayerId {
+        let name: &str = unsafe { CStr::from_ptr(__name).to_str().unwrap() };
+        let op: TournOp =
+            TournOp::RegisterPlayer(SquireAccount::new(name.to_string(), name.to_string()));
+
+        match FFI_TOURNAMENT_REGISTRY.get_mut().unwrap().get_mut(&self) {
+            Some(mut t) => {
```

Comment:
You shouldn't need to have `mut t` here. `mut` is for when the value of a variable changes, but `t` is a `&mut Tournament`. `mut t` implies that you are going to reference a different `Tournament`.
```suggestion
            Some(t) => {
```

Reply:
error[E0596]: cannot borrow `t` as mutable, as it is not declared as mutable
  --> squire_lib/src/ffi_tournament.rs:72:23
   |
71 |             Some(t) => {
   |                  - help: consider changing this to be mutable: `mut t`
72 |                 match t.apply_op(op) {
   |                       ^^^^^^^^^^^^^^ cannot borrow as mutable

For more information about this error, try `rustc --explain E0596`.
error: could not compile `squire_lib` due to previous error

Reply:
Oh, right! `DashMap::get_mut` doesn't return a reference. It returns a guard that gives out a reference.